### PR TITLE
`azurerm_web_application_firewall_policy` - Support `rule` for `rule_group_override`

### LINKED ISSUE #19347: Support for WAF Policy Rule Group Override
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

When configuring WAF policy, we need to change default action (AnomalyScoring) for default managed rules.

ARM support this in Microsoft.Network/ApplicationGatewayWebApplicationFirewallPolicies API version 2022-05-01, but terraform doesn't.

### New or Affected Resource(s)/Data Source(s)

azurerm_web_application_firewall_policy

### Potential Terraform Configuration

```hcl
resource "azurerm_web_application_firewall_policy" "example" {
  name                = "wafpolicy"
  resource_group_name = azurerm_resource_group.example.name
  location            = azurerm_resource_group.example.location

  managed_rules {
    managed_rule_set {
      type    = "OWASP"
      version = "3.2"

      rule_group_override {
        rule_group_name = "General"
        disabled_rules  = []
        rules = [
          rule {
            ruleId = "200004",
            state = "Enabled",
            action = "Log"
          }

          rule {
            ruleId = "200002",
            state = "Enabled",
            action = "Block"
          }
        ]
      }
    }
  }
}
```

### References

- https://learn.microsoft.com/en-us/azure/templates/microsoft.network/change-log/applicationgatewaywebapplicationfirewallpolicies
- https://learn.microsoft.com/en-us/azure/templates/microsoft.network/2022-05-01/applicationgatewaywebapplicationfirewallpolicies?pivots=deployment-language-bicep#managedruleoverride

### PR DESCRIPTION
**Pull request by ms-zhenhua**
fix https://github.com/hashicorp/terraform-provider-azurerm/issues/19347

image

### CODE REVIEW

File: internal/services/network/web_application_firewall_policy_resource.go [hunk 1]
```diff
@@ -272,11 +275,41 @@ func resourceWebApplicationFirewallPolicy() *pluginsdk.Resource {
 													Required:     true,
 													ValidateFunc: validate.ValidateWebApplicationFirewallPolicyRuleGroupName,
 												},
-												"disabled_rules": {
+												"rule": {
```
Comment (katbyte) [inline internal/services/network/web_application_firewall_policy_resource.go:278]: this is a breaking change?

**Review by katbyte — CHANGES_REQUESTED** [review]

File: internal/services/network/web_application_firewall_policy_resource.go (hunk 1 above)
Comment (ms-zhenhua) [reply internal/services/network/web_application_firewall_policy_resource.go:278]: No, `disabled_rules` is still kept in current version and will be replaced by `rule` in 4.0.  The `disabled_rules` is dynamically added at the end of this function controlled by `if !features.FourPointOhBeta()`.

Comment (katbyte) [reply internal/services/network/web_application_firewall_policy_resource.go:278]: I see that now! thanks

File: website/docs/r/web_application_firewall_policy.html.markdown [hunk 2]
```diff
@@ -226,9 +232,23 @@ The `managed_rule_set` block supports the following:
...
 The `rule_group_override` block supports the following:
 
-* `rule_group_name` - (Required) The name of the Rule Group
+* `rule_group_name` - (Required) The name of the Rule Group.
+
+* `rule` - (Optional) One or more `rule` block defined below.
+
+* `disabled_rules` - (Optional) One or more Rule IDs.
+
+-> **NOTE:** This property has been deprecated in favour of the `rule` property and will be removed in version 4.0 of the provider.
```
Comment (katbyte) [inline website/docs/r/web_application_firewall_policy.html.markdown:241]:
```suggestion
```

**Review by katbyte — APPROVED** [review]
Aside from 1 comment LGTM @ms-zhenhua 🦀

**Review by katbyte — CHANGES_REQUESTED** [review]
could we fix ci? @ms-zhenhua

**Comment by ms-zhenhua** [comment]
> could we fix ci? @ms-zhenhua

Seems the lint error is caused by https://github.com/hashicorp/terraform-provider-azurerm/pull/19569

**Comment by ms-zhenhua** [comment]
> could we fix ci? @ms-zhenhua

Hi @katbyte, the ci has been fixed. Kindly help review it again. Thanks. :blush:

**Review by katbyte — APPROVED** [review]
LGTM 🏷️

**Pull request merged**